### ISSUE TITLE
ComputeCanonicalTransform does not return correct principle axes

### ISSUE THREAD

**Issue by qzhu2017** [rdkit/rdkit#4375]
**Describe the bug**
I tried to use the `ComputeCanonicalTransform` function to return the principle axes. But the axes are not aligned correctly. 

**To Reproduce**
```python
from rdkit import Chem
from rdkit.Chem import AllChem
from rdkit.Chem import rdMolTransforms as rdmt
import numpy as np
from random import choice

for i in range(20):
    mol = Chem.MolFromSmiles('C1=C(C=C(C=C1[N+](=O)[O-])[N+](=O)[O-])C(=O)O')
    mol = Chem.AddHs(mol)
    AllChem.EmbedMultipleConfs(mol, numConfs=20, maxAttempts=200, useRandomCoords=True, pruneRmsThresh=0.5)
    N_confs = mol.GetNumConformers()
    id = choice(range(N_confs))
    conf = mol.GetConformer(id)
    trans = rdmt.ComputeCanonicalTransform(conf)
    print(i, np.linalg.det(trans[:3,:3]))
    if abs(abs(np.linalg.det(trans[:3,:3]))-1.0)>1e-1:
        print(trans)
        break
```
**Expected behavior**
The returned trans is a 4*4 matrix.
https://www.rdkit.org/docs/source/rdkit.Chem.rdMolTransforms.html#rdkit.Chem.rdMolTransforms.ComputeCanonicalTransform
```
0 -2.192973320416496e-15
[[-0.04144005  0.19475365  0.9799764   0.61016636]
 [-0.03073336  0.97919703 -0.20057079  1.00398005]
 [-0.04158121  0.19923934  0.97906828  0.61476988]
 [ 0.          0.          0.          1.        ]]
```
My understanding is that the first 3*3 matrix should be a rotation matrix with +1 or -1 determinant. However, it failed for my case.

**Configuration:**
 - RDKit version: rdkit-2021.03.4-py39h88273a1_0
 - OS: MAC-OS, CentOS
 - Python version (if relevant): Python 3.7/3.8
 - Are you using conda? Yes
 - If you are using conda, which channel did you install the rdkit from? conda-forge

**Comment by ptosco** [comment]
@qzhu2017 I’ll check this tonight as soon as I am back home.

**Comment by ptosco** [comment]
@qzhu2017 I have just run your test against the latest RDKit `master` branch and I got the expected results:
```
0 -1.0000000000000002
1 -1.0000000000000004
2 1.0000000000000004
3 -1.0
4 0.9999999999999999
5 -1.0000000000000002
6 0.9999999999999998
7 -1.0000000000000004
8 -1.0000000000000002
9 -0.9999999999999997
10 0.9999999999999999
11 1.0
12 1.0000000000000016
13 -1.0
14 1.0000000000000013
15 -1.0000000000000007
16 -1.0000000000000004
17 1.0000000000000002
18 1.0000000000000004
19 -1.0000000000000004
​```
I believe this is the same problem that I described in #4302 and that I have fixed in #4303. This fix is not yet part of RDKit release 2021.03.4.

**Comment by qzhu2017** [comment]
@ptosco I see. I am more than happy to run more tests. 
Do you know how I can install the latest version from the source code?

**Comment by qzhu2017** [comment]
@ptosco 
I tried to following this link to install a new RDKit.
https://www.rdkit.org/docs/Install.html#linux-x86-64-python-3-environment

I was able to finish the installation after a long wait. However, it does not have the AllChem module. Can you tell me if I missed anything in the installation? Do I need to activate something in the cmake?

**Comment by ptosco** [comment]
@qzhu2017 Things to check:
- `rm CMakeCache.txt` then re-run `cmake` and check that it picks up the correct Python executable and library (i.e., the same Python interpreter that you are using to run your script)
- check that the RDKit libraries that you built are in `<RDKit_root_dir>/lib`
- before running your script, make sure libraries and Python modules are in your path:
```
export RDBASE=<RDKit_root_dir>
export LD_LIBRARY_PATH=$RDBASE/lib:$LD_LIBRARY_PATH
export PYTHONPATH=$RDBASE:$PYTHONPATH
```

**Comment by qzhu2017** [comment]
I followed your suggestion and the problem is resolved. Thank you so much!


Just have a quick question. I compared the directory of `Chem` after my own build.

<img width="831" alt="image" src="https://user-images.githubusercontent.com/29445366/127728953-26e01fbb-981e-4482-accb-3c81988c6982.png">

Below is what I saw from Conda;
<img width="835" alt="image" src="https://user-images.githubusercontent.com/29445366/127728965-d2ab22f9-8e01-4879-a918-d50edb5acc6a.png">

It looks like my own build does not produce the `AllChem` and many other libraries. Is this expected?

**Comment by ptosco** [comment]
@qzhu2017 That’s the build directory. Type `make install`, then all your libraries will be installed in `rdkit/lib` and Python modules in `rdkit/RDKit` (i.e., one directory up from `build`)

**Comment by qzhu2017** [comment]
@ptosco Thank you. I am clear now. I suggest you add some more instructions to https://www.rdkit.org/docs/Install.html#linux-x86-64-python-3-environment

Mention that one can do parallel compilation, e.g. `make -j 20` if there are multiple cores available. Also, set up the environment as follows.
```
export RDBASE=<RDKit_root_dir>
export LD_LIBRARY_PATH=$RDBASE/lib:$LD_LIBRARY_PATH
export PYTHONPATH=$RDBASE:$PYTHONPATH
```

I confirm that new `ComputeCanonicalTransform` function works correctly. Thanks for your great help.

**Comment by ptosco** [comment]
@qzhu2017 Glad that it worked, thanks for letting me know. Closing the issue now.

**Comment by qzhu2017** [comment]
@ptosco 

I just tried to do 
`conda install -c conda-forge rdkit` 
to get the latest release of `v2021.03.5-` (updated about one month ago)
https://anaconda.org/conda-forge/rdkit/files
```
rdkit-2021.03.5            |   py39hccf6a74_0        38.3 MB  conda-forge
```

However, the script which I showed in the beginning of this post would fail if I used the conda version.

```python
from rdkit import Chem
from rdkit.Chem import AllChem
from rdkit.Chem import rdMolTransforms as rdmt
import numpy as np
from random import choice

for i in range(20):
    mol = Chem.MolFromSmiles('C1=C(C=C(C=C1[N+](=O)[O-])[N+](=O)[O-])C(=O)O')
    mol = Chem.AddHs(mol)
    AllChem.EmbedMultipleConfs(mol, numConfs=20, maxAttempts=200, useRandomCoords=True, pruneRmsThresh=0.5)
    N_confs = mol.GetNumConformers()
    id = choice(range(N_confs))
    conf = mol.GetConformer(id)
    trans = rdmt.ComputeCanonicalTransform(conf)
    print(i, np.linalg.det(trans[:3,:3]))
    if abs(abs(np.linalg.det(trans[:3,:3]))-1.0)>1e-1:
        print(trans)
        break

```
```
0 1.16945154430644e-15
[[-0.04140117  0.19502394  0.97992429  0.61061507]
 [-0.03077459  0.97913731 -0.20085582  1.00405668]
 [-0.04154299  0.19952437  0.97901185  0.61523438]
 [ 0.          0.          0.          1.        ]]
```


Has your fix been included in the conda-rdkit?

**Comment by ptosco** [comment]
@qzhu2017 The new Eigen3-based canonical transformation code s not yet part of 2021.03.5. I think it will be included in the forthcoming 2021.09 release which is due in a couple of weeks.
